internal/controller/factory/metadata.go

```diff
@@ -0,0 +1,22 @@
+package factory
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/aenix-io/etcd-operator/api/v1alpha1"
+)
+
+// mergeObjectMeta merges the metadata of an existing object with the metadata of a new object.
+// It copies the labels and annotations from the new object to the existing object.
+func mergeObjectMeta(old metav1.ObjectMeta, new v1alpha1.EmbeddedObjectMetadata) metav1.ObjectMeta {
```

Comment:
I would not name it merge if it just copies :)

Reply:
It's some sort of merge :) It merges objectMeta with specified embeddedObjectMetadata.

This function can be extended later with some more advanced logic (Such as propagation of EtcdCluster labels into all resources that we discussed earlier) So I suggest to leave this name

Reply:
It doesn't alter incoming parameters. It's a clean function. We deepcopy old medatata.